src/data/moves/pokemon-move.ts

```diff
@@ -0,0 +1,94 @@
+import type Pokemon from "#app/field/pokemon";
+import { toDmgValue } from "#app/utils/common";
+import type { MoveId } from "#enums/move-id";
+import { allMoves } from "../data-lists";
+import type Move from "./move";
+
+/**
+ * Wrapper class for the {@linkcode Move} class for Pokemon to interact with.
+ * These are the moves assigned to a {@linkcode Pokemon} object.
+ * It links to {@linkcode Move} class via the move ID.
+ * Compared to {@linkcode Move}, this class also tracks things like
+ * PP Ups recieved, PP used, etc.
+ * @see {@linkcode isUsable} - checks if move is restricted, out of PP, or not implemented.
+ * @see {@linkcode getMove} - returns {@linkcode Move} object by looking it up via ID.
+ * @see {@linkcode usePp} - removes a point of PP from the move.
+ * @see {@linkcode getMovePp} - returns amount of PP a move currently has.
+ * @see {@linkcode getPpRatio} - returns the current PP amount / max PP amount.
+ * @see {@linkcode getName} - returns name of {@linkcode Move}.
+ **/
+
```

Comment:
I wish VSCode didn't insert these random newlines when you use the "refactor" tool to move code to a different file.
```suggestion
 */
```